### PR TITLE
[MOBILE] Setup Firebase Crashlytics 

### PR DESCRIPTION
**Pull request by katendejericho5**
#### Summary of Changes (What does this PR do?)

-This sets up Fireabase Crashlytics in mobile 
#### Status of maturity (all need to be checked before merging):

- [x] I've tested this locally
- [ ] I consider this code done
- [ ] This change ready to hit production in its current state
- [ ] The title of the PR states what changed and the related issues number (used for the release note).
- [ ] I've included issue number in the "Closes #ISSUE-NUMBER" part of the "[What are the relevant tickets?](#what-are-the-relevant-tickets)" section to [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [ ] I've updated corresponding documentation for the changes in this PR.
- [ ] I have written unit and/or e2e tests for my change(s).

#### How should this be manually tested?

- Please include the steps to be done in order to setup and test this PR.
- 
- To test this 
- include a test crash 
- 
TextButton(
    onPressed: () => throw Exception(),
    child: const Text("Throw Test Exception"),
),
- then run the app, and then test to crash the app 
- then remove the recently added code and re-run the app , 
- in case there is no change ,
- , enable debug logging to see if your app is sending crash reports.
Enable and view debug logging for Crashlytics:
.........
Before running your app, set the following adb shell flag to DEBUG:

adb shell setprop log.tag.FirebaseCrashlytics DEBUG

View the logs in your device logs by running the following command:

adb logcat -s FirebaseCrashlytics

Then Force a test crash. 

Look for the following message or code 204 in your logcat output, either of which verifies that your app is sending crashes to Firebase.

**Crashlytics report upload complete**
After confirming that your app is sending crashes, you can optionally disable debug logging by setting the adb shell flag back to INFO:

adb shell setprop log.tag.FirebaseCrashlytics INFO
- 

#### What are the relevant tickets?

- Closes #<enter issue number here>
- [Jira_card_number]() (If exists)

### CODE REVIEW

**Comment by katendejericho5** [comment]
.

File: mobile/android/build.gradle [hunk 1]
```diff
@@ -9,6 +9,7 @@ buildscript {
         classpath 'com.android.tools.build:gradle:7.2.2'
         classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
         classpath 'com.google.gms:google-services:4.3.15'
+        classpath 'com.google.firebase:firebase-crashlytics-gradle:2.8.1'
```
Comment (NoahNsimbe) [inline mobile/android/build.gradle:12]: @katendejericho5 was this added by `flutterfire`??

File: mobile/lib/main.dart [hunk 2]
```diff
@@ -6,12 +6,15 @@ import 'package:app/widgets/widgets.dart';
...
-
+import 'package:firebase_crashlytics/firebase_crashlytics.dart';
 import 'firebase_options.dart';
 
 void main() async {
   WidgetsFlutterBinding.ensureInitialized();
+  await Firebase.initializeApp();
 
+// Pass all uncaught "fatal" errors from the framework to Crashlytics
+  FlutterError.onError = FirebaseCrashlytics.instance.recordFlutterFatalError;
```
Comment (NoahNsimbe) [inline mobile/lib/main.dart:17]: You can place this in the [initializeMainMethod()](https://github.com/airqo-platform/AirQo-frontend/blob/0c04447e5ef6878d9d4a3dae7a0387e79375047a/mobile/lib/main_common.dart#L124) method

**Review by NoahNsimbe — COMMENTED** [review]
Thanks, @katendejericho5. Left a few comments

**Review by NoahNsimbe — COMMENTED** [review]
Thanks, @katendejericho5. Left a few comments

File: mobile/android/build.gradle (hunk 1 above)
Comment (katendejericho5) [reply mobile/android/build.gradle:12]: Yes , it was added automatically

Comment (NoahNsimbe) [reply mobile/android/build.gradle:12]: Great. Thanks @katendejericho5

**Review by NoahNsimbe — APPROVED** [review]
LGTM Thanks @katendejericho5

**Comment by BenjaminSsempala** [comment]
Thanks @katendejericho5 . This looks good.
It might be helpful to structure the description, most especially the steps to test out. They are a little bit unclear.

**Review by BenjaminSsempala — APPROVED** [review]
LGTM @katendejericho5

**Pull request merged**